packages/react-router-dev/config/config.ts

```diff
@@ -736,28 +799,69 @@ export const ssrExternals = isInReactRouterMonorepo()
 function isInReactRouterMonorepo() {
   // We use '@react-router/node' for this check since it's a
   // dependency of this package and guaranteed to be in node_modules
-  let serverRuntimePath = path.dirname(
+  let serverRuntimePath = Path.dirname(
     require.resolve("@react-router/node/package.json")
   );
-  let serverRuntimeParentDir = path.basename(
-    path.resolve(serverRuntimePath, "..")
+  let serverRuntimeParentDir = Path.basename(
+    Path.resolve(serverRuntimePath, "..")
   );
   return serverRuntimeParentDir === "packages";
 }
 
+function omitRoutes(
+  config: ResolvedReactRouterConfig
+): ResolvedReactRouterConfig {
+  return {
+    ...config,
+    routes: {},
+  };
+}
+
 const entryExts = [".js", ".jsx", ".ts", ".tsx"];
 
+function isEntryFile(entryBasename: string, filename: string) {
+  return entryExts.some((ext) => filename === `${entryBasename}${ext}`);
+}
+
 function findEntry(
   dir: string,
   basename: string,
   options?: { absolute?: boolean }
 ): string | undefined {
   for (let ext of entryExts) {
-    let file = path.resolve(dir, basename + ext);
+    let file = Path.resolve(dir, basename + ext);
     if (fs.existsSync(file)) {
-      return options?.absolute ?? false ? file : path.relative(dir, file);
+      return options?.absolute ?? false ? file : Path.relative(dir, file);
     }
   }
 
   return undefined;
 }
+
+function isEntryFileDependency(
+  moduleGraph: Vite.ModuleGraph,
+  entryFilepath: string,
+  filepath: string
+): boolean {
+  // Ensure normalized paths
+  entryFilepath = Path.normalize(entryFilepath);
+  filepath = Path.normalize(filepath);
+
+  if (filepath === entryFilepath) {
+    return true;
+  }
+
+  let mod = moduleGraph.getModuleById(filepath);
+
+  if (!mod) {
+    return false;
+  }
+
+  for (let importer of mod.importers) {
+    if (importer.id === entryFilepath) {
+      return true;
+    }
+  }
```

Comment:
This seems to be checking for a new (direct) dependency of the entry file manually. Since we already invalidated the Vite Node context and asked for a new config, shouldn't the module graph be up-to-date, meaning that we can rely solely on the `moduleGraph.getModuleById`?

Reply:
No, this is actually handling the case where you're changing a file that's imported by the entry file. But looking at this again made me realise the logic isn't right so I've just pushed a fix.

Reply:
Ah ok got it, this is necessary because we want to detect _just_ route changes, so we want a module graph starting with `routes.ts` as the entrypoint. But what we have is a module graph starting with `react-router.config.ts` as the entrypoint, so this is crawling the subgraph that corresponds to the `routes.ts` module graph.